src/test/java/cat/udl/eps/softarch/demo/steps/GetTakeStepDefs.java

```diff
@@ -0,0 +1,32 @@
+package cat.udl.eps.softarch.demo.steps;
+
+import cat.udl.eps.softarch.demo.domain.Take;
+import cat.udl.eps.softarch.demo.repository.TakeRepository;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.When;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+
+import java.math.BigDecimal;
+import java.time.ZonedDateTime;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+
+public class GetTakeStepDefs {
+
+    @Autowired
+    private StepDefs stepDefs;
+
+    @Autowired
+    private TakeRepository takeRepository;
+
+    @When("^I get the Take$")
+    public void takeHasBeenCreated() throws Exception {
```

Comment:
You can reuse the step that tries to retrieve a take from other stepdefs, no need to repeat its implementation in different stepdefs

Reply:
Solved with @Oscars35 commits